src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -1277,11 +1278,17 @@ public List<SubscriptionDTO> getOwnerSubscriptions(String ownerKey) {
     public AsyncJobStatusDTO refreshPools(String ownerKey, Boolean autoCreateOwner) {
         Owner owner = ownerCurator.getByKey(ownerKey);
         if (owner == null) {
-            if (autoCreateOwner && ownerService.isOwnerKeyValidForCreation(ownerKey)) {
-                owner = createOwnerFromDTO(new OwnerDTO().key(ownerKey).displayName(ownerKey));
+            try {
+                if (autoCreateOwner && ownerService.isOwnerKeyValidForCreation(ownerKey)) {
+                    owner = createOwnerFromDTO(new OwnerDTO().key(ownerKey).displayName(ownerKey));
+                }
+                else {
+                    throw new NotFoundException(
+                        i18n.tr("owner with key: {0} was not found.", ownerKey));
+                }
             }
-            else {
-                throw new NotFoundException(i18n.tr("owner with key: {0} was not found.", ownerKey));
+            catch (OwnerInvalidKeyFormatException e) {
```

Comment:
`OwnerInvalidKeyFormatException` looks like it is intended for use by this Adapter code:

```java
  public boolean isOwnerKeyValidForCreation(String ownerKey) {
    try {
      LOGGER.debug("Checking if ownerkey={} is valid.", ownerKey);
      return userService.isOwnerKeyValid(Long.parseLong(ownerKey));
    } catch (NumberFormatException e) {
      LOGGER.debug("... ownerkey={} is invalid due to NumberFormatException.", ownerKey);
      throw new CandlepinAdapterException(ownerKey + " is not a valid OrgId", e);
    }
  }
```

I don't believe I ever was happy about the exception getting thrown by Adapters here. To me, it should have simply returned `false`, because that is the answer to the `isOwnerKeyValidForCreation` question.

With that in mind, are you okay if we get rid of OwnerInvalidKeyFormatException, and Adapters can not throw an exception here, and return `false` instead?

Reply:
That seems reasonable to me